cirq-core/cirq/circuits/circuit.py

```diff
@@ -908,6 +909,10 @@ def qid_shape(
         qids = ops.QubitOrder.as_qubit_order(qubit_order).order_for(self.all_qubits())
         return protocols.qid_shape(qids)
 
+    @deprecated(deadline='v0.13', fix='use all_measurement_key_names instead')
```

Comment:
Assuming `cirq.Circuit` methods are the most heavily used in Cirq, is a ` v0.13` deprecation deadline a little soon?

Reply:
After `v0.13` we want to repurpose `all_measurement_keys` to return `MeasurementKey`s instead of `str`, similar to the `measurement_key` protocol. So, I think it would be nice to keep those in sync. After the deadline, the return type will be different and the fix will be simply enveloping it in `str()` if they want it to keep working with existing code.

Reply:
:+1: